### PR TITLE
Removing user namespace on user delete

### PR DESCRIPTION
**Pull request by nathan-jenan-rancher**
This change adds logic to the user lifecycle to remove the user namespace
when a user is deleted.

Issue:
https://github.com/rancher/rancher/issues/12576

### CODE REVIEW

File: pkg/controllers/management/auth/user.go [hunk 1]
```diff
@@ -4,31 +4,38 @@ import (
...
+	tokens           v3.TokenInterface
+	namespaces       v1.NamespaceInterface
+	prtbLister       v3.ProjectRoleTemplateBindingLister
+	crtbLister       v3.ClusterRoleTemplateBindingLister
+	grbLister        v3.GlobalRoleBindingLister
+	prtbIndexer      cache.Indexer
+	crtbIndexer      cache.Indexer
+	grbIndexer       cache.Indexer
+	tokenIndexer     cache.Indexer
+	namespaceIndexer cache.Indexer
```
Comment (cjellick) [inline pkg/controllers/management/auth/user.go]: You dont need an indexer, since you just need to get by name. you can just use the lister. `lister.Get()`

Comment (nathan-jenan-rancher) [reply pkg/controllers/management/auth/user.go]: Duh, thank you!

File: pkg/controllers/management/auth/user.go [hunk 2]
```diff
@@ -308,3 +321,21 @@ func (l *userLifecycle) deleteAllTokens(tokens []*v3.Token) error {
 
 	return nil
 }
+
+func (l *userLifecycle) deleteUserNamespace(username string) error {
+	namespace, err := l.namespaceLister.Get("", username)
+	if err != nil {
```
Comment (cjellick) [inline pkg/controllers/management/auth/user.go]: One of the errors can be an not exists or not found error (i forget which). if we hit that error, can you _not_ bubble it up?

**Comment by cjellick** [comment]
LGTM

**Pull request merged**